lib/teleterm/gateway/gateway.go

```diff
@@ -148,14 +193,12 @@ func (g *Gateway) CLICommand() (string, error) {
 
 // Gateway describes local proxy that creates a gateway to the remote Teleport resource.
 type Gateway struct {
-	Config
-
+	cfg        *Config
```

Comment:
Previously any caller could change any public field on `Config` which would actually be detrimental to the main subject of this PR. We don't want people to be able to change the underlying `LocalPort` without actually restarting the gateway.

This enables us to encapsulate this logic within a setter method rather than depending on the promise that we'll remember to always restart the gateway after changing `gateway.LocalPort`.